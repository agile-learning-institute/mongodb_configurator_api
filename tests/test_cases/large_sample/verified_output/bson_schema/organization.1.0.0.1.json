{
  "description": "An organization in the system",
  "bsonType": "object",
  "properties": {
    "_id": {
      "description": "The unique identifier for the organization",
      "bsonType": "objectId"
    },
    "name": {
      "description": "The organization name",
      "pattern": "^[^\\t\\n\\r]{0,255}$",
      "bsonType": "string"
    },
    "status": {
      "description": "The current status of the organization",
      "bsonType": "string",
      "enum": [
        "active",
        "archived"
      ]
    },
    "last_saved": {
      "description": "When the organization document was last updated",
      "bsonType": "object",
      "properties": {
        "from_ip": {
          "description": "Http Request remote IP address",
          "pattern": "^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
          "bsonType": "string"
        },
        "by_user": {
          "description": "ID Of User",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        },
        "at_time": {
          "description": "The date-time when last updated",
          "bsonType": "date"
        },
        "correlation_id": {
          "description": "The logging correlation ID of the update transaction",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        }
      },
      "additionalProperties": false,
      "required": [
        "from_ip",
        "by_user",
        "at_time",
        "correlation_id"
      ]
    },
    "users": {
      "description": "Users associated with this organization",
      "bsonType": "array",
      "items": {
        "description": "A user identifier",
        "bsonType": "objectId"
      }
    },
    "website": {
      "description": "Organization website",
      "pattern": "^https?://[^\\s]+$",
      "bsonType": "string"
    },
    "home_address": {
      "description": "Organization home address",
      "bsonType": "object",
      "properties": {
        "street": {
          "description": "Street address",
          "pattern": "^[^\\t\\n\\r]{0,255}$",
          "bsonType": "string"
        },
        "city": {
          "description": "City",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        },
        "state": {
          "description": "State or province",
          "pattern": "^[A-Z]{2}$",
          "bsonType": "string"
        },
        "postal_code": {
          "description": "Postal code",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        }
      },
      "additionalProperties": false,
      "required": [
        "street",
        "postal_code"
      ]
    }
  },
  "required": [
    "_id",
    "name",
    "status",
    "last_saved"
  ],
  "additionalProperties": false
}