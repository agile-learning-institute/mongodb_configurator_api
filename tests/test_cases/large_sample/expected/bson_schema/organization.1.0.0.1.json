{
  "title": "Organization",
  "description": "An organization in the system",
  "bsonType": "object",
  "properties": {
    "_id": {
      "description": "The unique identifier for the organization",
      "bsonType": "objectId"
    },
    "name": {
      "description": "The organization name",
      "bsonType": "string",
      "pattern": "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
    },
    "status": {
      "description": "The current status of the organization",
      "bsonType": "string",
      "enum": ["active", "archived"]
    },
    "last_saved": {
      "title": "Breadcrumb",
      "description": "When the organization document was last updated",
      "bsonType": "object",
      "properties": {
        "from_ip": {
          "description": "Http Request remote IP address",
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "by_user": {
          "description": "ID Of User",
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "at_time": {
          "description": "The date-time when last updated",
          "bsonType": "date"
        },
        "correlation_id": {
          "description": "The logging correlation ID of the update transaction",
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        }
      },
      "additionalProperties": false,
      "required": ["from_ip", "by_user", "at_time", "correlation_id"]
    },
    "users": {
      "description": "Users associated with this organization",
      "bsonType": "array",
      "items": {
        "description": "A user identifier",
        "bsonType": "objectId"
      }
    },
    "website": {
      "description": "Organization website",
      "bsonType": "string",
      "pattern": "^https?://[\\w\\d\\-]+(\\.[\\w\\d\\-]+)+([\\w\\d\\-._~:/?#\\[\\]@!$&'()*+,;=]*)?$"
    },
    "home_address": {
      "title": "Street Address",
      "description": "Organization home address",
      "bsonType": "object",
      "properties": {
        "street": {
          "description": "Street address",
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
        },
        "city": {
          "description": "City",
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "state": {
          "description": "State or province",
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "postal_code": {
          "description": "Postal code",
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        }
      },
      "additionalProperties": false,
      "required": ["postal_code"]
    }
  },
  "additionalProperties": false,
  "required": ["_id", "name", "status", "last_saved"]
}
